### PR TITLE
[gfx90a] explicity disable asm solvers

### PR DESCRIPTION
**Pull request by Slimakanzer**
This PR disables asm solvers for gfx90a and marks as _WORKAROUND_ for #1146.

### CODE REVIEW

File: src/solver/conv_asm_implicit_gemm_bwd_v4r1_dynamic.cpp [hunk 1]
```diff
@@ -133,6 +135,10 @@ bool ConvAsmImplicitGemmV4R1DynamicBwd::IsApplicable(const ConvolutionContext& c
         return false;
 
     const auto device_name = ctx.GetStream().GetDeviceName();
+#if WORKAROUND_ISSUE_1146
+    if(device_name == "gfx90a")
```
Comment (shaojiewang) [inline src/solver/conv_asm_implicit_gemm_bwd_v4r1_dynamic.cpp:139]: In fact due to line 142, if device name is gfx90a, this solver will return false. So the W/A might not be necessary.

Comment (junliume) [reply src/solver/conv_asm_implicit_gemm_bwd_v4r1_dynamic.cpp:139]: +1, it really depends on how explicitly you want it to be (and for later reference to get back to this issue). While technically I agree that it is redundant.

File: src/solver/conv_asm_implicit_gemm_v4r1_dynamic.cpp [hunk 2]
```diff
@@ -278,6 +280,10 @@ bool ConvAsmImplicitGemmV4R1DynamicFwd::IsApplicable(const ConvolutionContext& c
         return false;
 
     const auto device_name = ctx.GetStream().GetDeviceName();
+#if WORKAROUND_ISSUE_1146
```
Comment (junliume) [inline src/solver/conv_asm_implicit_gemm_v4r1_dynamic.cpp:283]: Similarly, redundant conditions due to line 281/287?

File: src/solver/conv_asm_implicit_gemm_wrw_v4r1_dynamic.cpp [hunk 3]
```diff
@@ -300,6 +302,10 @@ bool ConvAsmImplicitGemmV4R1DynamicWrw::IsApplicable(const ConvolutionContext& c
         return false;
 
     const auto device_name = ctx.GetStream().GetDeviceName();
+#if WORKAROUND_ISSUE_1146
```
Comment (junliume) [inline src/solver/conv_asm_implicit_gemm_wrw_v4r1_dynamic.cpp:305]: Similar to previous comment.

File: src/solver/conv_bin_winoRxS.cpp [hunk 4]
```diff
@@ -246,6 +248,11 @@ bool ConvBinWinogradRxS::IsApplicable(const ConvolutionContext& params) const
 
     const auto name = params.GetStream().GetDeviceName();
     const bool fp16 = params.IsFp16();
+
+#if WORKAROUND_ISSUE_1146
```
Comment (junliume) [inline src/solver/conv_bin_winoRxS.cpp:252]: Similar to previous comments

File: src/solver/conv_asm_implicit_gemm_bwd_v4r1_dynamic.cpp (hunk 1 above)
Comment (atamazov) [reply src/solver/conv_asm_implicit_gemm_bwd_v4r1_dynamic.cpp:139]: Yes, this should be reverted as we have other Solvers for devices that have MAI (and that is why this Solver is not enabled for MI100). #1146 was updated recently.

File: src/solver/conv_asm_implicit_gemm_v4r1_dynamic.cpp (hunk 2 above)
Comment (atamazov) [reply src/solver/conv_asm_implicit_gemm_v4r1_dynamic.cpp:283]: +1

File: src/solver/conv_asm_implicit_gemm_wrw_v4r1_dynamic.cpp (hunk 3 above)
Comment (atamazov) [reply src/solver/conv_asm_implicit_gemm_wrw_v4r1_dynamic.cpp:305]: +1

File: src/solver/conv_bin_winoRxS.cpp (hunk 4 above)
Comment (atamazov) [reply src/solver/conv_bin_winoRxS.cpp:252]: @junliume This Solver is enabled on gfx908. We are going to to enable it on gfx90a as well. Even if technically is not necessary, I would better keep it here as a reminder.

@Slimakanzer [Recommendation] Rework this code in order to make the W/A really necessary ;)

**Review by atamazov — CHANGES_REQUESTED** [review]
Please resolve:
- https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1150#pullrequestreview-752153033
- https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1150#pullrequestreview-752308274
- https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1150#pullrequestreview-752308409

Please also look at https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1150#pullrequestreview-752308704 and decide yourself what to do.

**Comment by atamazov** [comment]
@shaojiewang Thanks for reviewing!

File: src/solver/conv_bin_winoRxS.cpp (hunk 4 above)
Comment (Slimakanzer) [reply src/solver/conv_bin_winoRxS.cpp]: @atamazov 
Thank you! Reverted this unnecessary restriction (see [explanation](https://github.com/ROCmSoftwarePlatform/MIOpen/issues/1146#issuecomment-919021192))

**Review by atamazov — APPROVED** [review]
LGTM!

**Pull request merged**